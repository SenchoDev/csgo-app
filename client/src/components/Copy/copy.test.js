import React from 'react';
import { shallow } from 'enzyme';
import { Copy } from './copy.component';

describe('Copy component', () => {
  let wrapper;
  let mockToggleCopiedButton;
  beforeEach(() => {
    mockToggleCopiedButton = jest.fn();

    const mockProps = {
      toggleCopiedButton: mockToggleCopiedButton,
      crosshair: 'crosshairData',
    }

    wrapper = shallow(<Copy {...mockProps}/>)
  })

  it('should render Copy component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleCopiedButton to true if component is clicked', () =>{
    wrapper.find('CopyToClipboard').simulate('click');
    expect(mockToggleCopiedButton).toBeTruthy();
  })
})

