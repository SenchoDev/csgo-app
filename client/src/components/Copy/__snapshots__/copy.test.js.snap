// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy component should render CartIcon component 1`] = `ShallowWrapper {}`;

exports[`Copy component should render Copy component 1`] = `ShallowWrapper {}`;
